src/transformers/azrael/main.py

```diff
@@ -0,0 +1,170 @@
+                    self._on_rented(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "Lent":
+                    self._on_lent(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "Returned":
+                    self._on_returned(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "LendingStopped":
+                    self._on_lending_stopped(decoded_params, tx_hash)
+                elif event["decoded"]["name"] == "CollateralClaimed":
+                    self._on_collateral_claim(decoded_params, tx_hash)
+                
+            logging.info(event)
+
+        self._flush_state = True
+
+    # todo: should be part of the interface
+    # todo: acts as the means to sync with db state
+    def update_memory_state(self) -> None:
+        """_summary_"""
+
+        if len(self._transformed) > 0:
+            return
+
+        state = self._db.get_all_items(self._db_name, self._collection_name)
+
+        if state is None:
+            return
+
+        self._transformed = state
+
+    # todo: should be part of the interface
+    def flush(self) -> None:
+        """_summary_"""
+
+        if self._flush_state:
+            # * write to the db
+            self._db.put_items(self._transformed, self._db_name, self._collection_name)
+            self._flush_state = False
+
+    def _on_collateral_claim(self, decoded_params, tx_hash: str) -> None:
+        # CollateralClaimed(indexed uint256 lendingId, uint32 claimedAt)
+
+        self._transformed.append(Event('CollateralClaimed', tx_hash, _COLLATERAL_CLAIMED_FIELDS, decoded_params))
+
+    def _on_lending_stopped(self, decoded_params, tx_hash: str) -> None:
+        # LendingStopped(indexed uint256 lendingId, uint32 stoppedAt)
+
+        self._transformed.append(Event('LendingStopped', tx_hash, _LENDING_STOPPED_FIELDS, decoded_params))
+
+    def _on_returned(self, decoded_params, tx_hash: str) -> None:
+        # Returned(indexed uint256 lendingId, uint32 returnedAt)
+
+        self._transformed.append(Event('Returned',tx_hash,  _RETURNED_FIELDS, decoded_params))
+
+    def _on_rented(self, decoded_params, tx_hash: str) -> None:
+        # Rented(uint256 lendingId, indexed address renterAddress, uint8 rentDuration, uint32 rentedAt)
+
+        self._transformed.append(Event('Rented', tx_hash, _RENTED_FIELDS, decoded_params))
+
+    def _on_lent(self, decoded_params, tx_hash: str) -> None:
+        # Lent(indexed address nftAddress, indexed uint256 tokenId, uint8 lentAmount, uint256 lendingId,
+        # indexed address lenderAddress, uint8 maxRentDuration, bytes4 dailyRentPrice,
+        # bytes4 nftPrice, bool isERC721, uint8 paymentToken)
+
+        event = Event('Lent', tx_hash, _LENT_FIELDS, decoded_params)
+        event['dailyRentPrice'] = unpack_price(event['dailyRentPrice'])
+        event['nftPrice'] = unpack_price(event['nftPrice'])
+
+        self._transformed.append(event)
+
+
+def hex_to_int(s):
```

Comment:
this looks like general enough to be moved somewhere outside of here

Reply:
If we do that we will lose the package modularity since sylvester and azrael will need to share files.

we could have a utils.py inside src>transformers>shared .. ?

I would prefer to have duplicate code but preserve the 100% modularity we currently have. Let me know which you want.